docs/troubleshooting.asciidoc

```diff
@@ -128,4 +128,18 @@ set ELASTIC_APM_STARTUP_HOOKS_LOGGING=1
 
 and then running the application in a context where the environment variable will be visible. In setting this value,
 an `ElasticApmAgentStartupHook.log` file is written to in the directory containing the startup hook assembly, in addition to
-writing to standard output.
\ No newline at end of file
+writing to standard output.
+
+[float]
+[[agent-overhead]]
+=== The agent causes too much overhead
+
+A good place to start is our <<config-all-options-summary>>. There are multiple settings there with the `Performance` keyword which help you to tweak the agent for your needs.
+
+The most expensive operation in the agent is typically stack trace capturing. The agent, by default only captures stack traces for spans with a duration of 5ms or more, and with a limit of 50 stack frames.
+
```

Comment:
```suggestion
```

The sentences before and after this line rely on each other. I'd remove this break. You also might consider bulleting out your suggestions and leading with the outcome instead of the configuration variable. In other words, something like this:

> **The agent causes too much overhead**
> A good place to start is [config-all-options-summary]. There are multiple settings there with the `Performance` keyword which can help you tweak the agent for your needs.
> 
> The most expensive operation in the agent is typically stack trace capturing. The agent, by default, only captures stack traces for spans with a duration of 5ms or more, and with a limit of 50 stack frames.
> If this is too much in your environment, consider disabling stack trace capturing either partially or entirely:
> 
> * To disable stack trace capturing for spans, but continue to capture stack traces for errors, set the [config-span-frames-min-duration] to `0` and leave the [config-stack-trace-limit] on its default.
> * To disable stack trace capturing entirely–which in most applications reduces the agent overhead dramatically–set [config-stack-trace-limit] to `0`.